userprofile/tests/test_models.py

```diff
@@ -122,3 +123,17 @@ def test_get_or_prefill_existing_healthchecks(self):
         self.assertEqual(profile.first_name, "oldfirst")
         self.assertEqual(profile.last_name, "newlast")
         self.assertEqual(profile.preexisting_condition, "no")
+
+    @responses.activate
+    @override_settings(TBCONNECT_GROUP_ARM_ACTIVE=True)
```

Comment:
We can maybe also test the opposite, if study is disabled don't populate the field and if field has a value don't update it again.